### ISSUE TITLE
Is there a multiple line breaks support?

### ISSUE THREAD

**Issue by ozgrozer** [erusev/parsedown#597]
Is it possible to make something like this?

**input:**

```
first line




sixth line
```

**output:**

```
<p>first line</p>
<p></p>
<p></p>
<p></p>
<p></p>
<p>sixth line</p>
```

or

```
<p>first line</p>
<br>
<br>
<br>
<br>
<p>sixth line</p>
```

**Comment by aidantwoods** [comment]
Have you tried the following? :)
```php
$parsedown = new Parsedown;
$parsedown->setBreaksEnabled(true);
```

**Comment by aidantwoods** [comment]
Right, sorry – I think this is the second time I've forgotten that `setBreaksEnabled` only honours a single line break 😬 

#450 I think asks the same question, the solution there was essentially to create an extension doing the equivalent of 
 this diff https://github.com/aidantwoods/parsedown/commit/fd681ee683a1e17f0e2afd75909d7dabf453a1b7#diff-ed0d3da57330712681e64f838087ea47.

That said, it involved patching the `lines` method (which is now mostly contained in `linesElements`) – which you'd need to keep up to date with each release of Parsedown. For this reason (and since this is now the third time I think this has been asked: see also #448) it might be better to add this in as an optional feature in Parsedown itself since it touches an area that doesn't lend itself well to extension changes.

I expect to merge this into the release of 1.8.0 – since this is a new feature I'll keep the PR open for at least a couple days incase anyone wants to raise any concerns about the implementation though :)

**Comment by ozgrozer** [comment]
It works good but there's a bug.

If I give this input:

```
first line




sixth line

---
```

I get this output:

```
<h2>
first line
<br>
<br>
<br>
<br>
<br>
sixth line
<br>
</h2>
```

**Comment by aidantwoods** [comment]
Thanks for reporting this.

While I could fix the first issue, the second is harder to fix using the method in which this is implemented because this method essentially removes the ability for consecutive newlines to interrupt a paragraph (as mentioned in https://github.com/erusev/parsedown/issues/450#issuecomment-267959893).

To address the table issue, the way in which Parsedown currently works is to transform an existing paragraph into the table header. Parsedown doesn't expect a paragraph to contain sequential blank lines because these should interrupt a paragraph. In-fact, Parsedown shouldn't be rendering a table at all in this case (since a table should not interrupt a paragraph of more than one line – which is fixed in #612).

---

When thinking about how to adjust the current method so that the second example works, it would require adding logic into several block handlers to change their behaviours based on the `setLiteralBreaks` setting.

I also remembered there was an issue with line breaks getting lost that affects default functionality #609, and so #610 fixes that by counting how many interrupts occur (instead of just noting one exists).

Using the counting of interrupts, in combination with some block overloading semantics to be added in #611, the `linesElements` method no longer needs to be adjusted, and so this is definitely within the realm of being implemented sensibly by an extension. For this reason #611 also removes the `setLiteralBreaks` feature that was due to be added (since the main reason for planning to add the feature was because it would require overloading a complex method – that's no longer the case).

---

In order to get a table to render you'd need some way of ending a paragraph before the table (since Parsedown can't distinguish the table header from the contents before it if multiple newlines do not end the paragraph).

The following should be a minimal example of an extension that will add the behaviour you want:
```php
class LiteralBreaks extends Parsedown
{
    protected function paragraph($Line)
    {
        if (substr($Line['text'], -1) === '#')
        {
            $closed = true;

            $Line['text'] = substr($Line['text'], 0, -1);
        }

        $Block = parent::paragraph($Line);

        if (isset($closed))
        {
            $Block['closed'] = true;
        }

        return $Block;
    }

    protected function paragraphContinue($Line, array $Block)
    {
        if (isset($Block['closed']))
        {
            return;
        }

        if (isset($Block['interrupted']))
        {
            $Block['element']['handler']['argument'] .= str_repeat("\n  ", $Block['interrupted']);
            unset($Block['interrupted']);
        }

        if (substr($Line['text'], -1) === '#')
        {
            $Block['closed'] = true;

            $Line['text'] = substr($Line['text'], 0, -1);
        }

        $Block['element']['handler']['argument'] .= "\n".$Line['text'];

        return $Block;
    }
}

```

In the above, I've added the closing character `#` so that if a line ends with `#` then the paragraph will close to help address the table problem.

Thus, if you wish to forcibly end a paragraph in the same way that newlines would normally do, just end a line with a `#`.

You might want to look though the `block*Continue` methods to make sure they account for interrupts when they occur where desired.

The given extension will render the following:

```gfm
first line




sixth line

---
**Table:**#

First Header | Second Header
--- | ---
Content cell | Content cell
Content cell | Content cell
```

as:

```html
<p>first line
<br />
<br />
<br />
<br />
sixth line</p>
<hr />
<p><strong>Table:</strong></p>
<table>
<thead>
<tr>
<th>First Header</th>
<th>Second Header</th>
</tr>
</thead>
<tbody>
<tr>
<td>Content cell</td>
<td>Content cell</td>
</tr>
<tr>
<td>Content cell</td>
<td>Content cell</td>
</tr>
</tbody>
</table>
```
